onadata/apps/logger/models/attachment.py

```diff
@@ -86,3 +87,12 @@ def file_hash(self):
     def filename(self):
         if self.media_file:
             return os.path.basename(self.media_file.name)
+
+    def soft_delete(self, user=None):
+        """
+        Soft deletes an attachment by adding a deleted_at timestamp.
+        """
+        self.deleted_at = timezone.now()
+        if user is not None:
```

Comment:
Would this work if we kept `if user:` here ?

Reply:
Sure. I'll change that.